src/Analyzer.JsonRuleEngine.UnitTests/JsonRuleEngineTests.cs

```diff
@@ -338,5 +340,89 @@ private string CreateRulesFromEvaluationDefinitions(string[] ruleEvaluationDefin
+        [DataRow(@"{
+                ""inclusions"": {
+                        ""severity"": [3]
+                    }
+            }", 9, new int[] { 3 }, DisplayName = "Only Severity 3")]
+        [DataRow(@"{
+                ""exclusions"": {
+                        ""severity"": [3]
+                    }
+            }", 19, null, null, new int[] { 3 }, DisplayName = "Excludes Severity 3")]
+        [DataRow(@"{
+                ""inclusions"": {
+                        ""ids"": [""TA-000001""]
+                    }
+            }", 1, null, new string[] { "TA-000001" }, DisplayName = "Only Id TA-000001")]
+        [DataRow(@"{
+                ""exclusions"": {
+                        ""ids"": [""TA-000001""]
+                    }
+            }", 27, null, null, null, new string[] { "TA-000001" }, DisplayName = "All rules except Id TA-000001")]
+        [DataRow(@"{
+                ""inclusions"": {
+                        ""severity"": [3],
+                        ""ids"": [""TA-000001""]
+                    }
+            }", 10, new int[] { 3 }, new string[] { "TA-000001" }, DisplayName = "Only Severity 3 and  Id TA-000001")]
+        [DataRow(@"{
+                ""inclusions"": {
+                        ""ids"": [""TA-000001""]
+                    }, 
+                ""severityOverrides"": {
+                    ""TA-000001"": 3
+                    }
+                }", 1, null, new string[] { "TA-000001" }, null, null, DisplayName = "Only Id TA-000001, change Severity to 3")]
+        public void FilterRules_ValidInputValues_ReturnCorrectFilteredRules(string configuration, int expectedEvaluationCount, 
+            IEnumerable<Severity> includeSeverities = null, IEnumerable<string> includeIds = null, 
+            IEnumerable<Severity> excludeSeverities = null, IEnumerable<string> excludeIds = null)
+        {
+            // Arrange
+            var jsonRuleEngine = JsonRuleEngine.Create(TemplateAnalyzer.LoadRules(), t => null);
+
+            // Filter
+            jsonRuleEngine.FilterRules(configuration);
+
+            // Compare
+            Assert.AreEqual(expectedEvaluationCount, jsonRuleEngine.RuleDefinitions.Count);
+
+            if (includeSeverities != null || includeIds != null)
+            {
+                foreach (var r in jsonRuleEngine.RuleDefinitions)
+                {
+                    Assert.IsTrue((includeSeverities != null && includeSeverities.Contains(r.Severity)) ||
+                        (includeIds != null && includeIds.Contains(r.Id)));
+                }
+            }
+            if (excludeSeverities != null || excludeIds != null)
+            {
+                foreach (var r in jsonRuleEngine.RuleDefinitions)
+                {
+                    Assert.IsTrue((excludeSeverities != null && !excludeSeverities.Contains(r.Severity)) ||
+                        (excludeIds!= null && !excludeIds.Contains(r.Id)));
+                }
+            }
+        }
```

Comment:
Please also add tests for *severityOverrides*

Reply:
I did already - test called "Only Id TA-000001, change Severity to 3"

Reply:
Ok - but the test never actually validates the severity of the rule anywhere.  In other words, *severityOverrides* could do nothing, and the test would still pass.